### PR TITLE
Refactor TrueFalseHold, move TrueHold to Obsolete

### PR DESCRIPTION
**Pull request by AntoineGautier**
This addresses #3787 (as well as #3917, and incidentally #3841).

In addition, the following blocks were slightly refactored due to the suppression of event iteration in the refactored hold block (see the revision notes of these blocks for more details).

- `Buildings.Templates.Plants.Controls.StagingRotation.StageCompletion`
- `Buildings.Templates.Plants.Controls.Utilities.HoldReal`
- `Buildings.Controls.OBC.ASHRAE.G36.AHUs.MultiZone.VAV.Validation.Controller` and `Buildings.Controls.OBC.ASHRAE.G36.AHUs.MultiZone.VAV.Validation.Controller_UnspecifiedClimate`: Both models use a hold to break the algebraic loop between `conAHU.y1SupFan` and `conAHU.u1SupFan`. As explained in https://github.com/lbl-srg/modelica-buildings/issues/3883, holding a signal does not fundamentally avoid direct feedback. The models are refactored with a sampler to emulate the supply fan status.

TODO before merging:

- [x] #3917
- [x] #3915
- [x] #3906
- [x] #3883
- [x] Wait for Modelon's feedback on the use of `pre(u)=u` as an initial equation [Modelon - 1263]: OCT complains about a singular initial equation system in some cases (https://app.travis-ci.com/github/lbl-srg/modelica-buildings/jobs/622920064#L434-L438), whereas Dymola and OMC handle these cases without any warning.
  - This is indeed a bug in OCT. A protected parameter `pre_u_start` has been introduced as a temporary workaround. It shall be removed when [Modelon - 1263] is fixed, and the initial equation changed back to `pre(u)=u` (which complies with MLS).

### CODE REVIEW

**Comment by AntoineGautier** [comment]
@mwetter @JayHuLBL This is (finally!) ready. Could one of you do the review?

File: Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo [hunk 1]
```diff
@@ -115,11 +115,20 @@ model Controller "Validation controller model"
...
-    final trueHoldDuration=1)
-    "Break loop"
-    annotation (Placement(transformation(extent={{220,30},{240,50}})));
-
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToReal booToRea(
+    final realTrue=1,
+    final realFalse=0)
+    "Convert to real"
+    annotation (Placement(transformation(extent={{190,70},{170,90}})));
+  Buildings.Fluid.BaseClasses.ActuatorFilter fil(
```
Comment (JayHuLBL) [inline Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo:123]: We should not use block that is not in the `OBC` package.

File: Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller_UnspecifiedClimate.mo [hunk 2]
```diff
@@ -122,11 +122,20 @@ model Controller_UnspecifiedClimate
...
-    final trueHoldDuration=1)
-    "Break loop"
-    annotation (Placement(transformation(extent={{220,30},{240,50}})));
-
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToReal booToRea(
+    final realTrue=1,
+    final realFalse=0)
+    "Convert to real"
+    annotation (Placement(transformation(extent={{190,70},{170,90}})));
+  Buildings.Fluid.BaseClasses.ActuatorFilter fil(
```
Comment (JayHuLBL) [inline Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller_UnspecifiedClimate.mo:130]: Same as above.

File: Buildings/Obsolete/Controls/OBC/CDL/Logical/TrueHold.mo [hunk 3]
```diff
@@ -1,4 +1,4 @@
-within Buildings.Controls.OBC.CDL.Logical;
+within Buildings.Obsolete.Controls.OBC.CDL.Logical;
```
Comment (JayHuLBL) [inline Buildings/Obsolete/Controls/OBC/CDL/Logical/TrueHold.mo:1]: add an anotation as `obsolete = "This model is obsolete"`, see the block `Buildings.Obsolete.Controls.OBC.CDL.Logical.TriggeredTrapezoid`

**Review by JayHuLBL — COMMENTED** [review]
@AntoineGautier see the inline comments.

File: Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller_UnspecifiedClimate.mo (hunk 2 above)
Comment (AntoineGautier) [reply Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller_UnspecifiedClimate.mo]: Addressed with afb1c60

File: Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo (hunk 1 above)
Comment (AntoineGautier) [reply Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo]: Addressed with afb1c60

File: Buildings/Obsolete/Controls/OBC/CDL/Logical/TrueHold.mo (hunk 3 above)
Comment (AntoineGautier) [reply Buildings/Obsolete/Controls/OBC/CDL/Logical/TrueHold.mo:1]: Addressed with afb1c60

**Comment by AntoineGautier** [comment]
@JayHuLBL Thanks for the review just before your vacation! I addressed your comments and I think this is now ready to merge.

**Review by JayHuLBL — APPROVED** [review]

File: Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo [hunk 4]
```diff
@@ -120,15 +120,12 @@ model Controller "Validation controller model"
...
-  Buildings.Fluid.BaseClasses.ActuatorFilter fil(
-    final f=5/(2*Modelica.Constants.pi*10),
-    initType=Modelica.Blocks.Types.Init.InitialOutput,
-    final y_start=0)
-    "Filter signal"
-    annotation (Placement(transformation(extent={{160,70},{140,90}})));
   Buildings.Controls.OBC.CDL.Reals.GreaterThreshold greThr(final t=0.5)
     "Compare filtered signal to threshold to trigger true status"
     annotation (Placement(transformation(extent={{130,70},{110,90}})));
+  Buildings.Controls.OBC.CDL.Discrete.ZeroOrderHold sam(samplePeriod=10)
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo:126]: @AntoineGautier 
This change is not good. It is implemented as
image
with a 10 second sampling time. If this were used in a model with an HVAC system, simulation would be very slow due to the frequent sampling (and users are likely to copy paste such fragments from examples).

Can we use the same patterns as we used last week for the enable signal of the HP templates? Or just use a direct feedthrough. Even a 2nd order filter would be better here I think as the 10 second sampling is very frequent and may also give issues in cross-tool verification.

**Review by mwetter — CHANGES_REQUESTED** [review]
Please see the inline comment

File: Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo (hunk 4 above)
Comment (AntoineGautier) [reply Buildings/Controls/OBC/ASHRAE/G36/AHUs/MultiZone/VAV/Validation/Controller.mo]: This was refactored in fc5e57b9cf using a timer.
As explained in https://github.com/lbl-srg/modelica-buildings/issues/3923#issuecomment-2202418450 (which I updated with the alternative implementation using a timer), the implementation using a timer does not systematically break algebraic loops.
However, in the validation models in which it is used here, it accomplishes this goal without using classes outside the CDL package, as Jianjun suggested.

**Review by mwetter — APPROVED** [review]

**Pull request merged**